tests/src/Search/ElasticTestCase.php

```diff
@@ -0,0 +1,158 @@
+use Psr\Log\NullLogger;
+use Silex\WebTestCase;
+use Symfony\Component\BrowserKit\Client;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\StreamOutput;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+abstract class ElasticTestCase extends WebTestCase
+{
+    protected $isLocal;
+    protected $console;
+    /** @var Kernel */
+    protected $kernel;
+    /** @var ElasticsearchClient */
+    protected $client;
+    /** @var Client */
+    protected $api;
+
+    public function getJsonResponse()
+    {
+        $response = $this->getResponse();
+        if (!$response->isOk()) {
+            $this->fail('Response returned was not 200');
+        }
+        $json = json_decode($response->getContent());
+
+        return $json;
+    }
+
+    public function newClient()
+    {
+        return $this->api = static::createClient();
+    }
+
+    public function getResponse()
+    {
+        return $this->api->getResponse();
+    }
+
+    public function addDocumentToElasticSearch(string $doc)
+    {
+        $obj = json_decode($doc);
+        $this->client->indexJsonDocument($obj->type, $obj->id, $doc, true);
+    }
+    public function addDocumentsToElasticSearch(array $docs)
+    {
+        foreach ($docs as $doc) {
+            $this->addDocumentToElasticSearch($doc);
+        }
+    }
+
+    public function createConfiguration()
+    {
+        if (file_exists(__DIR__.'/../../../config/local.php')) {
+            $this->isLocal = true;
+            $config = include __DIR__.'/../../../config/local.php';
+        } else {
+            $this->isLocal = false;
+            $config = include __DIR__.'/../../../config/ci.php';
+        }
+
+        return $config;
+    }
+
+    protected function jsonRequest(string $verb, string $endpoint, array $params = array())
+    {
+        $headers = array(
+            'HTTP_ACCEPT' => 'application/json',
+            'CONTENT_TYPE' => 'application/json',
+        );
+
+        return $this->api->request(
+            $verb,
+            $endpoint,
+            $params,
+            [],
+            $headers
+        );
+    }
+
+    /**
+     * Creates the application.
+     *
+     * @return HttpKernelInterface
+     */
+    public function createApplication()
+    {
+        $this->kernel = new Kernel($this->createConfiguration());
+
+        return $this->kernel->getApp();
+    }
+
+    public function getElasticSearchClient(callable $fn = null): ElasticsearchClient
+    {
+        return $fn ? $fn($this->kernel->get('elastic.client')) : $this->kernel->get('elastic.client');
+    }
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->client = $this->getElasticSearchClient();
+        $lines = $this->runCommand('search:setup');
+        if ($lines[0] === 'No alive nodes found in your cluster') {
+            $this->markTestSkipped('Elasticsearch may not be installed, skipping');
```

Comment:
I've changed it to a fail for now, if thats also works? I've got these tests in a "web" group so locally we can do `phpunit --exclude-group=web`